apps/lifxlights/lifx_lights.star

```diff
@@ -0,0 +1,437 @@
+    """
+    token = config.str("token")
+    light_id = config.str("light_id")
+    blink = config.bool("blink", DEFAULT_BLINK)
+
+    # if there's no token or light id, render the demo
+    if token == None or light_id == None:
+        return render_demo_setup()
+
+    # render error message when no lights are found during config
+    if light_id == "nolights":
+        return render_error("No lights found", ":(")
+
+    # render error message when listing the lights during config fails
+    if light_id == "listlighterror":
+        return render_error("Error listing lights", "Check token")
+
+    # try to get the light information
+    rep = http.get(LIFX_URL + "/lights/id:" + light_id, headers = {
+        "Authorization": "Bearer " + token,
+    })
+
+    # render error if API fails
+    if rep.status_code != 200:
+        error = rep.json()["error"] or "Error calling LIFX API"
+        return render_error(error, get_error_message_for_status_code(rep.status_code))
+
+    light = rep.json()[0]
+
+    hue = light["color"]["hue"]
+    saturation = light["color"]["saturation"]
+    brightness = light["brightness"]
+    brightness_percent = int(brightness * 100)
+    power = light["power"].upper()
+
+    rgb = hsb_to_rgb(hue, saturation, brightness)
+
+    return render.Root(
+        child = render.Box(
+            child = render.Column(
+                main_align = "space_evenly",
+                cross_align = "center",
+                children = [
+                    render.Image(src = LIFX_LOGO, height = 10),
+                    render.Marquee(
+                        width = 60,
+                        align = "center",
+                        child = render.Padding(pad = (0, 1, 0, 0), child = render.Text(light["label"], font = "5x8")),
+                    ),
+                    render.Row(
+                        expanded = False,
+                        main_align = "center",
+                        cross_align = "center",
+                        children = [
+                            render.Padding(pad = (2, 1, 2, 0), child = get_circle_widget(rgb, power, blink)),
+                            render.Padding(pad = (2, 1, 2, 0), child = render.Text(power)),
+                            render.Padding(pad = (0, 1, 0, 0), child = render.Text("(%d%%)" % brightness_percent if power == "ON" else "")),
+                        ],
+                    ),
+                ],
+            ),
+        ),
+    )
+
+def get_schema():
+    """
+    Returns the app configuration schema. Asks for the LIFX
+    personal access token first, and then generates the next
+    options dynamically.
+
+        Returns:
+            schema (schema): The configuration schema.
+    """
+
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "token",
+                name = "Personal Access Token",
+                desc = "Token generated at https://cloud.lifx.com.",
```

Comment:
These descriptions are actually not visible to users in the iOS/Android app at the moment. Probably want to add the same information to the app description to help folks along.